lib/test_prof/before_all.rb

```diff
@@ -15,9 +15,10 @@ def initialize
     class << self
       attr_accessor :adapter
 
-      def begin_transaction
+      def begin_transaction(ctx)
```

Comment:
@palkan, This bit deserves your attention. I made the hook run in the same context as `before_all` gets in rspec. I think I did the same in minitest, but haven't tested that.

Reply:
What is a purpose of passing a context and executing a hook within it?
Isn't `setup_before_all` meant for _global state_ management?

Reply:
@palkan, is there a good reason not to allow access to instance variables?

Reply:
Providing a context means that there's a path forward towards reducing global state.

Reply:
> Providing a context means that there's a path forward towards reducing global state.

I see the opposite: by allowing to manipulate the example group state from the global hook could introduce a global dependency adding too many responsibility to `before_all`. That means, that with the ability to _hack_ the context, we can lose the interoperability between `before_all` with `before`.

That seems that we're trying to re-invent the shared contexts.

Let's remove the context-handling and simplify the hooks. That would be enough for the Discourse case, and we can discuss the extended API later.

Reply:
@palkan, OK, done.

Reply:
Thanks!